src/simulation/Simulation.cpp

```diff
@@ -5139,13 +5140,44 @@ void Simulation::SetCustomGOL(std::vector<CustomGOLData> newCustomGol)
 	customGol = newCustomGol;
 }
 
-String Simulation::ElementResolve(int type, int ctype) const
+String Simulation::ElementResolve(int type, int ctype, int tmp, int tmp4, bool includeCtype) const
 {
-	if (type == PT_LIFE)
+	String displayedType = "Empty";
+	uint8_t ctypeMode = 0;
+	uint8_t tmpMode = 0;
+	bool pipeLike = false;
+
+	if (IsElement(type))
+	{
+		displayedType = elements[type].Name;
+		ctypeMode = elements[type].HudProperties & HUD_CTYPE_BITS;
+		tmpMode = elements[type].HudProperties & HUD_TMP_BITS;
+		pipeLike = elements[type].HudProperties & HUD_PIPELIKE;
+	}
+
+	if (tmpMode == HUD_TMP_FILTMODE)
+	{
+		if(!includeCtype)
+			return displayedType;
+
+		String filtModes[] = {"set colour", "AND", "OR", "subtract colour", "red shift", "blue shift", "no effect", "XOR", "NOT", "old QRTZ scattering", "variable red shift", "variable blue shift"};
+		String filtMode = "unknown mode";
+		if (tmp >= 0 && tmp <= 11)
+			filtMode = filtModes[tmp];
+		return String::Build(displayedType, " (", filtMode, ")");
+	}
+	else if (ctypeMode == HUD_CTYPE_PREFIXED && IsElement(ctype) )
+	{
+		String element = elements[ctype].Name;
+		String prefix = language::Template{ "DEFAULT_LS_GAME_HUD_PREFIX_"+displayedType.ToUpper() }();
```

Comment:
This is what I meant. The template should be parametrized with `displayedType`. Also, I'm not sure the `ToUpper`. Needs some looking into. In any case, the language string ID shouldn't come from the Name property, but the element identifier.